src/frontend/apps/lti_site/components/VideoWidgetProvider/widgets/DownloadVideo/index.tsx

```diff
@@ -69,6 +94,61 @@ export const DownloadVideo = () => {
     [intl, messages],
   );
 
+  const [toggleAllowDownload, setToggleAllowDownload] = useState(
+    video.show_download,
+  );
+
+  const onSuccessCallback = useCallback(
+    (videoUpdated: Video) => {
+      toast.success(
+        intl.formatMessage(
+          videoUpdated.show_download
+            ? messages.allowDownloadToggleSuccess
+            : messages.disallowDownloadToggleSuccess,
+        ),
+        {
+          position: 'bottom-center',
+        },
+      );
+    },
+    [
+      toast,
+      intl,
+      messages.disallowDownloadToggleSuccess,
+      messages.allowDownloadToggleSuccess,
+    ],
+  );
+
+  const onErrorCallback = useCallback(
+    (err) => {
+      setToggleAllowDownload(video.show_download);
+      report(err);
+      toast.error(intl.formatMessage(messages.allowDownloadToggleFail), {
+        position: 'bottom-center',
+      });
+    },
+    [
+      toast,
+      intl,
+      report,
+      video.show_download,
+      messages.allowDownloadToggleFail,
+    ],
+  );
+
+  const videoMutation = useUpdateVideo(video.id, {
+    onSuccess: onSuccessCallback,
+    onError: onErrorCallback,
+  });
+
+  const onToggleChange = useCallback(() => {
+    setToggleAllowDownload(!video.show_download);
```

Comment:
you can move it to the `onSuccess` and while the mutation is not over you can use the `mutation.isLoading` to disable the toggle button. If there is an error, the state is not changed and you don't have to manage it in the `onError`